client.go

```diff
@@ -433,3 +433,118 @@ func (e GenericOpenAPIError) Body() []byte {
 func (e GenericOpenAPIError) Model() interface{} {
 	return e.model
 }
+
+// Below are the custom Mux API Error types, so that users of the SDK can identify
+// what issue they're running into more easily.
+
+// Generic 400 error
+type BadRequestError struct {
+	body  []byte
+	error string
+}
+
+func (e BadRequestError) Error() string {
+	return e.error
+}
+
+// 401 Error
+type UnauthorizedError struct {
+	body  []byte
+	error string
+}
+
+func (e UnauthorizedError) Error() string {
+	return e.error
+}
+
+func (e UnauthorizedError) Body() []byte {
+	return e.body
+}
+
+// 403 Error
+type ForbiddenError struct {
+	body  []byte
+	error string
+}
+
+func (e ForbiddenError) Error() string {
+	return e.error
+}
+
+func (e ForbiddenError) Body() []byte {
+	return e.body
+}
+
+// 404 Error
+type NotFoundError struct {
+	body  []byte
+	error string
+}
+
+func (e NotFoundError) Error() string {
+	return e.error
+}
+
+func (e NotFoundError) Body() []byte {
+	return e.body
+}
+
+// 429 Error
+type TooManyRequestsError struct {
+	body  []byte
+	error string
+}
+
+func (e TooManyRequestsError) Error() string {
+	return e.error
+}
+
+func (e TooManyRequestsError) Body() []byte {
+	return e.body
+}
+
+// 5XX Error
+type ServiceError struct {
+	body   []byte
+	status int
+	error  string
+}
+
+func (e ServiceError) Error() string {
+	return e.error
+}
+
+func (e ServiceError) Body() []byte {
+	return e.body
+}
+
+func (e ServiceError) Code() int {
+	return e.status
+}
+
+// Helper to check for common non-200 errors
+func CheckForHttpError(code int, body []byte) error {
+
+	if code >= 200 && code <= 299 {
+		return nil
+	}
+
+	switch code {
```

Comment:
on the other side of this, callers will have to handle this as a type switch if they care to know what kind of error was returned. the only other approach I see that might make sense is to allow checking on the `Error()` which isn't necessarily idiomatic either. I illustrated  that kind of an approach here - https://play.golang.org/p/CTFDWgQXiAB for your consideration...

Reply:
I'm flexible, but this doesn't feel _too_ bad to me:

```
    fmt.Println("Getting dead asset...")
    _, err := client.AssetsApi.GetAsset("foo")

    if err != nil {
        switch err.(type) {
        case muxgo.NotFoundError:
            fmt.Println("Not Found")
        case muxgo.BadRequestError:
            fmt.Println("Bad Request")
        default:
            fmt.Println(":shrug:")
        }
    }
```